src/Sylius/Bundle/AdminBundle/Twig/Ux/ComponentTemplateFinder.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\AdminBundle\Twig\Ux;
+
+use Symfony\UX\TwigComponent\ComponentTemplateFinderInterface;
+use Twig\Loader\LoaderInterface;
+
+final readonly class ComponentTemplateFinder implements ComponentTemplateFinderInterface
```

Comment:
I don't think there's anything that's stopping us from making this a pretty generic service, just pass the prefix and templates directory through the constructor and it could be reused anywhere.